src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1075,6 +1079,45 @@ internal OneOrMany<SyntaxTree> GetSyntaxTreesByMappedPath(string mappedPath)
             }
         }
 
+        private class ContentHashComparer : IEqualityComparer<ReadOnlyMemory<byte>>
+        {
+            public bool Equals(ReadOnlyMemory<byte> x, ReadOnlyMemory<byte> y)
+            {
+                return x.Span.SequenceEqual(y.Span);
+            }
+
+            public int GetHashCode(ReadOnlyMemory<byte> obj)
+            {
+                return Hash.GetFNVHashCode(obj.Span, out _);
```

Comment:
No need to hash again. Xxhash128 is already mixed well.  Just grab the first 4 bytes.

Reply:
Done